### ISSUE TITLE
Map hash near the top of  a replay file sometimes doesn't correspond to the correct map

### ISSUE THREAD

**Issue by gdavegdave** [OpenRA/OpenRA#13450]
The map hash recorded near the top of replay files sometimes doesn't correspond to the map played on. For example, in this replay:
  https://www.gamereplays.org/openra/replays.php?game=92&show=details&id=320073
the map hash toward the beginning of the file is (found, in this case, using 'less'):
        Map: 58da5c480d13a00a07c9bee00f928f70564f188f

However, the game was played on the map with hash: 
  a03d9bfa8211e6d5233b4734ae40eefe52b19de1
That hash appears later on in the replay file and OpenRA does realise which map the game was actually played on.

I suspect that this occurs when the map was changed in the lobby. It's been a minor inconvenience for me as I've often looked to see which map a game was played on by 'less'ing the top of the replay file. 

It doesn't seem to serve any purpose to have the hashes of other maps stored in a replay file. So, I wonder, would this be trivial to fix? Ideally, the correct map hash should be available near the top of the replay file. If that wouldn't be trivial to achieve then I'd leave it as is.

**Comment by pchote** [comment]
The replay is a record of the network traffic with a metadata footer tacked on the end.  This is a case of "you're reading it wrong", and can't (and shouldn't!) be fixed.

**Comment by gdavegdave** [comment]
Yeah, fair enough. I had started filing the 'bug' when I realised that the replay *must* have the correct hash somewhere... and something along the lines of what you said. 

It's just proved a pain when trying to find the map of a random replay (I've resorted to opening the game, screen grabbing the message saying the map isn't available, cropping to the hash, and uploading to an OCR website). Well, now I know that I just need to delve deeper into the replay file or wait for r0b0v's feature of having the map be imported automatically.
